File: 0009-palindrome-number/0009-palindrome-number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        #any negative values are false
        if x < 0:
            return False
        div = 1
        
        #divider is the one used to get the left most number
        #get the divider
        while x >= 10*div:
            div *= 10
        while x:
            # r = x % 10 get the right digit
            # l = x // div get the left digit
            
            if x%10 != x//div:
                return False
            x = (x%div)//10
            div /= 100
        return True

File: 0009-palindrome-number/0009-palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) 
    {
        int newNumber =0 ,result,temp = x;
        if (x < 0)
        x *= -1;
        
        while(x!=0)
        {
            result = x%10;
            newNumber = newNumber*10 + result;
            x = x /10;
        }  

        if (temp == newNumber)
            return true;
        else
            return false;
    }
}